change_analyzer/wrappers/sequence_recorder.py

```diff
@@ -61,11 +61,33 @@ def get_enriched_page_source(info: WebDriver) -> str:
             - x, y coordinates
             - h, w dimensions
         """
-        # Get elements' by xpath from given page source
-        root = html.fromstring(info.page_source)
+        try:
+            page_source = info.page_source
+        except:
+            # Driver is no longer available - no page source to use
+            return ''
+
+        if 'xml' in page_source:
+            root = html.fromstring(page_source.encode("utf-16"))
+        else:
+            root = html.fromstring(info.page_source)
+
         tree = root.getroottree()
-        all_elements_by_xpath = root.xpath('//*')
-        for element in all_elements_by_xpath:
+
+        # Get elements' by xpath from given page source and
+        all_elements_from_root = root.xpath('//*')
+        all_elements_from_driver = info.find_elements_by_xpath("//*")
+
+        if 'xml' in page_source:
+            for index, el in enumerate(all_elements_from_root[2::]):
+                # We bypass the first two elements of the list (/html and /html/body) - they have no map in the driver
+                el.set('x', f"{all_elements_from_driver[index].location['x']}")
+                el.set('y', f"{all_elements_from_driver[index].location['y']}")
+                el.set('height', f"{all_elements_from_driver[index].size['height']}")
+                el.set('width', f"{all_elements_from_driver[index].size['width']}")
+            return etree.tostring(root).decode("utf-8")

```

Comment:
Above `root` is encoded with UTF-16 but here it is decoded with UTF-8. Can that cause problems?